### ISSUE TITLE
Centreon : UNKNOWN: SNMP Table Request: Cant get a single value. Switch HP Procurve 1920s-24G

### ISSUE THREAD

**Issue by TucoSBC** [centreon/centreon-plugins#1737]
Hi everyone,

First of all, i'm French then I hope you'll understand my bad English and forgive me !

I have a problem with my switch HP Procurve 1920s-24G on Centreon 19.10 (i was with 18.10 before and it didn't works too). I use the plugin network::hp::procurve::snmp::plugin and the modes CPU and Memory don't work at all, i have the unknown message : UNKNOWN: SNMP Table Request: Cant get a single value.

I have, in my Centreon, other switch HP procurve but more recent, 2510G-48 and 2530-24G and it works... With the same command

CPU = ./centreon_plugins.pl --plugin=network::hp::procurve::snmp::plugin --mode=cpu --hostname=10.X.X.X --snmp-community=mycommunity --snmp-version=1 --warning=90 --critical=80 --snmp-autoreduce

MEMORY = ./centreon_plugins.pl --plugin=network::hp::procurve::snmp::plugin --mode=memory --hostname=10.X.X.X --snmp-community=mycommunity --snmp-version=1 --warning=90 --critical=80 --snmp-autoreduce

If someone have a solution for my problem, i would be very grateful !

TucoSBC

**Comment by garnier-quentin** [comment]
Could you provide me a snmpwalk ? 
snmpwalk -ObentU -v 2c -c public IP .1.3.6.1.4.1 > hp_procurve.snmpwalk

**Comment by TucoSBC** [comment]
Excuse me but what is -ObentU, I did your command but it block, i change -v 2c in -v 1 and -c public in my private community.

**Comment by TucoSBC** [comment]
I tried to do an snmpwalk -c mycmnty -v 1 @IP .1.3.6.1.4.1

And there are unlimited lines

**Comment by TucoSBC** [comment]
[root@centreon-central mode]# snmpwalk -v 1 -c mycommunity IP .1.3.6.1.4.1
SNMPv2-SMI::enterprises.11.2.36.1.1.2.6.0 = STRING: "PD.02.09"
SNMPv2-SMI::enterprises.11.2.36.1.1.2.8.0 = STRING: "U-Boot 2012.10-00118-g3773021 (Oct 11 2016 - 15:39:54)"
SNMPv2-SMI::enterprises.11.2.36.1.1.2.9.0 = STRING: "CN70K3Q4NX"
SNMPv2-SMI::enterprises.11.2.36.1.2.1.0 = INTEGER: 2
SNMPv2-SMI::enterprises.11.2.36.1.3.1.0 = INTEGER: 0
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.1.0 = STRING: "HPE OfficeConnect Switch 1920S 24G 2SFP PoE+ (370W) JL385A, PD.02.09, Linux 3.6.5-375bd0e8, U-Boot 2012.10-00118-g3773021 (Oct 11 2016 - 15:39:54)"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.2.0 = STRING: "HPE OfficeConnect Switch 1920S 24G 2SFP PoE+ (370W) JL385A"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.3.0 = STRING: "hpe1920S_24G_PoE+"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.4.0 = STRING: "CN70K3Q4NX"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.6.0 = STRING: "A"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.7.0 = STRING: "BCM53344"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.8.0 = STRING: "0xbc00"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.9.0 = Hex-STRING: D0 67 26 48 19 D8
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.10.0 = STRING: "Linux 3.6.5-375bd0e8"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.11.0 = STRING: "BCM53344_A0"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.12.0 = STRING: " QOS"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.1.13.0 = STRING: "PD.02.09"
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.2.1.0 = INTEGER: 67
SNMPv2-SMI::enterprises.11.5.7.5.7.1.1.1.1.2.3.0 = INTEGER: 0

I keep the first lines that seems to be the more useful to share

**Comment by garnier-quentin** [comment]
I need the complete result.

**Comment by TucoSBC** [comment]
[hp_procurve_snmp_walk.txt](https://github.com/centreon/centreon-plugins/files/3863865/hp_procurve_snmp_walk.txt)
There it is

**Comment by TucoSBC** [comment]
Can I give it to you by mail?

**Comment by TucoSBC** [comment]
Sorry I blunder, for CPU the message is UNKNOWN: SNMP GET Request : Cant get a single value.
and for memory UNKNOWN: SNMP Table Request: Cant get a single value.

I don't know if it changes something but I prefer to precise it

**Comment by garnier-quentin** [comment]
Yes you can sent it to my emai

**Comment by TucoSBC** [comment]
It's done

**Comment by TucoSBC** [comment]
Another information, the mode environment doesn't works too, it show the result : All 0 components are all ok [0/0].

All the three modes I use, maybe the configuration of this 3 modes aren't adapted to this switch ?

**Comment by TucoSBC** [comment]
Any answer ?

**Comment by garnier-quentin** [comment]
Indeed. I need mibs file for that equipment. OIDs are in branch 'hpMoonshot-45G-Switch-Module'. But i don't find the mib file on internet for that branch: .1.3.6.1.4.1.11.5.7.5.7.1

**Comment by TucoSBC** [comment]
Ok i'll search from my side too

**Comment by TucoSBC** [comment]
Do you find something since the last time ?

**Comment by garnier-quentin** [comment]
No. I need mibs files.

**Comment by TucoSBC** [comment]
But where could we find this ?

**Comment by TucoSBC** [comment]
http://mibs.snmplabs.com/asn1/HP-PROCURVE-420-PRIVATE-MIB

That's it ?

**Comment by garnier-quentin** [comment]
It's not the right mib. It's the branch: .1.3.6.1.4.1.11.5.7.5.7.1 (hpMoonshot-45G-Switch-Module)

**Comment by garnier-quentin** [comment]
I have found the MIB on HP website: https://support.hpe.com/hpesc/public/docDisplay?docId=c04531771&docLocale=en_US#N101F5

**Comment by garnier-quentin** [comment]
In that MIB, i can get CPU and memory. 
What is the hp chassis ? HP Moonshot 1500 Chassis ? In your snmpwalk (not complete), it's missing the branch: .1.3.6.1.4.1.11.5.7.5.7.1.1.1.43

**Comment by garnier-quentin** [comment]
You can test it with plugin: ```--plugin=network/hp/moonshot/snmp/plugin.pm```

**Comment by mronell2** [comment]
and where to add this MIB in centreon, I have the same problem
